tfchain_bridge/pkg/stellar/stellar.go

```diff
@@ -327,79 +313,52 @@ func (w *StellarWallet) MonitorBridgeAccountAndMint(ctx context.Context, mintFn
 			if effect.GetAccount() != w.config.StellarBridgeAccount {
 				continue
 			}
-			if effect.GetType() == "account_credited" {
-				creditedEffect := effect.(horizoneffects.AccountCredited)
-				if creditedEffect.Asset.Code != asset[0] && creditedEffect.Asset.Issuer != asset[1] {
+
+			if effect.GetType() != "account_credited" {
+				continue
+			}
+
+			creditedEffect := effect.(horizoneffects.AccountCredited)
+			if creditedEffect.Asset.Code != asset[0] && creditedEffect.Asset.Issuer != asset[1] {
+				continue
+			}
+
+			ops, err := w.getOperationEffect(tx.Hash)
+			if err != nil {
+				continue
+			}
+
+			for _, op := range ops.Embedded.Records {
+				if op.GetType() != "payment" {
 					continue
 				}
-				parsedAmount, err := amount.ParseInt64(creditedEffect.Amount)
-				if err != nil {
+
+				paymentOpation := op.(operations.Payment)
+				if paymentOpation.To != w.config.StellarBridgeAccount {
 					continue
 				}
 
+				parsedAmount, err := amount.ParseInt64(paymentOpation.Amount)
+				if err != nil {
+					continue
+				}
 				depositedAmount := big.NewInt(int64(parsedAmount))
 
-				err = mintFn(tx.Account, depositedAmount, tx.Hash)
+				err = mintFn(paymentOpation.From, depositedAmount, tx)
```

Comment:
since the tx hash is saved on chain, a tx with 2 payment operations will only have 1 of them executed (as they result in 2 mints on chain, one will be rejected). So this would mean that first, you need to create a map with the accounts and how much they should receive based on all the payments, and then, this map needs to be passed to the extrinsic, so a single tx (with possibly multiple operations) can be executed in 1 tx on chain as well